### ISSUE TITLE
Interference from windows.h about std::max in problem.h

### ISSUE THREAD

**Issue by zhongjingjogy** [PatWie/CppNumericalSolvers#108]
A similar issue is reported [here](https://stackoverflow.com/questions/5004858/stdmin-gives-error).

It is suggested that 
```
(std::min)(a, b); // added parentheses around function name
(std::max)(a, b);
```

**Comment by PatWie** [comment]
I would prefer `std::min<Dtype>` then. Does `#define NOMINMAX` work for you?

**Comment by PatWie** [comment]
fixed in ec0a32e1ce3d299157e94b82a88a5530c5a93502
